test/ResourceCacheSpec.js

```diff
@@ -135,5 +135,27 @@ describe('ResourceCache', function() {
         }
       });
     });
+
+    describe('when a uuid is provided', function() {
```

Comment:
I now understand what is going on in this function after looking at this test.  looks good.